app/lib/dartdoc/backend.dart

```diff
@@ -265,8 +266,35 @@ class DartdocBackend {
     await _deleteAllWithPrefix(prefix, concurrency: concurrency);
   }
 
+  /// Schedules the garbage collection of the [package] and [version].
+  /// The wait queue is in-memory, it is not persisted.
+  void scheduleGC(String package, String version) {
+    _gcTasks.add(_GCTask(package, version));
+  }
+
+  /// Runs obsolete GC of old dartdoc files with low overhead (concurrency = 1).
+  ///
+  /// The function never returns.
+  Future<void> processScheduledGCTasks() async {
+    for (;;) {
+      if (_gcTasks.isEmpty) {
+        await Future.delayed(Duration(seconds: 30));
+        continue;
+      }
+      final task = _gcTasks.first;
+      _gcTasks.remove(task);
```

Comment:
Nit, maybe `final task _gcTasks.removeAt(0);`

Reply:
`_gcTasks` is a set, it has no `removeAt(int)`.